### ISSUE TITLE
Failed SSL Handshake (Android)

### ISSUE THREAD

**Issue by ken-kentan** [TooTallNate/Java-WebSocket#599]
## Current Behavior
Failed to SSL Handshake when connecting to [SSL server](https://github.com/TooTallNate/Java-WebSocket/blob/master/src/main/example/SSLServerExample.java).
But, can connect some other SSL server. like, `wss://echo.websocket.org`

## Steps to Reproduce (for bugs)
1. Launch a [SSL server](https://github.com/TooTallNate/Java-WebSocket/blob/master/src/main/example/SSLServerExample.java) at '8887' port.
2. Launch a [SSL client](https://github.com/TooTallNate/Java-WebSocket/blob/master/src/main/example/SSLClientExample.java) on Android.
3. Try connect.

## Client Debug log (for bugs)
```
javax.net.ssl.SSLHandshakeException: Handshake failed
    at com.android.org.conscrypt.OpenSSLSocketImpl.startHandshake(OpenSSLSocketImpl.java:429)
    at com.android.org.conscrypt.OpenSSLSocketImpl.waitForHandshake(OpenSSLSocketImpl.java:682)
    at com.android.org.conscrypt.OpenSSLSocketImpl.getInputStream(OpenSSLSocketImpl.java:644)
    at org.java_websocket.client.WebSocketClient.run(WebSocketClient.java:260)
    at java.lang.Thread.run(Thread.java:761)
Caused by: javax.net.ssl.SSLProtocolException: SSL handshake aborted: ssl=0xaa19cb80: Failure in SSL library, usually a protocol error
error:100000f7:SSL routines:OPENSSL_internal:WRONG_VERSION_NUMBER (external/boringssl/src/ssl/tls_record.c:192 0xa706d196:0x00000000)
    at com.android.org.conscrypt.NativeCrypto.SSL_do_handshake(Native Method)
    at com.android.org.conscrypt.OpenSSLSocketImpl.startHandshake(OpenSSLSocketImpl.java:357)
	... 4 more
```

## Your Environment
* Version used: 1.3.5
* Java version: 1.8.0_151-b12
* Operating System and version: Android7.1(Client), Windows10(Server)

**Comment by ken-kentan** [comment]
added description.

**Comment by ken-kentan** [comment]
I tested on emulated Android device.

I could connect to server with old Android (less API 23).
But, newer Android (API 23~) did not.

Maybe, I think the SSL WebSocketServer using not supported cipher suite in newer Android.

**Comment by ken-kentan** [comment]
Resolved #540

**Comment by marci4** [comment]
Hello @ken-kentan,
the SSLWebsocketServer is using the enabled cipher suites and protocols.
It depends on your android version which are enabled and therefore active!

Looks like it could be SSLV3, which may have caused the issue! There are many threads on stackoverflow because of that!

Happy you managed to solve it on your own!
Next time please include the description right away then I can take a look on it quicker/when I have less time on my hands!

Greetings
marci4
